### PR TITLE
stop using `assertObjectHasAttribute()`/`assertObjectHasNotAttribute()`

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | related #49233
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by GromNaN — APPROVED** [review]
For reference: https://github.com/sebastianbergmann/phpunit/issues/4601

File: src/Symfony/Bridge/Doctrine/Tests/ManagerRegistryTest.php [hunk 1]
```diff
@@ -44,7 +44,8 @@ public function testResetService()
         $registry->resetManager();
 
         $this->assertSame($foo, $container->get('foo'));
-        $this->assertObjectNotHasAttribute('bar', $foo);
+        $this->assertInstanceOf(\stdClass::class, $foo);
+        $this->assertTrue(!isset($foo->bar));
```
Comment (derrabus) [inline src/Symfony/Bridge/Doctrine/Tests/ManagerRegistryTest.php]: Not sure if that makes a difference here, but the more precise check would be `property_exists()`. `$foo->bar = null;` would have been caught by the old assertion, but it passes the `!isset()` check.

Comment (xabbuh) [reply src/Symfony/Bridge/Doctrine/Tests/ManagerRegistryTest.php]: indeed, makes sense

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Bridge/Doctrine/Tests/ManagerRegistryTest.php [hunk 2]
```diff
@@ -44,7 +44,8 @@ public function testResetService()
         $registry->resetManager();
 
         $this->assertSame($foo, $container->get('foo'));
-        $this->assertObjectNotHasAttribute('bar', $foo);
+        $this->assertInstanceOf(\stdClass::class, $foo);
+        $this->assertTrue(!property_exists($foo, 'bar'));
```
Comment (phansys) [inline src/Symfony/Bridge/Doctrine/Tests/ManagerRegistryTest.php]:
```suggestion
        $this->assertFalse(property_exists($foo, 'bar'));
```

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**